dojo/unittests/tools/test_gitlab_dast_parser.py

```diff
@@ -0,0 +1,95 @@
+from django.test import TestCase
+from dojo.tools.gitlab_dast.parser import GitlabDastParser
+from dojo.models import Test
+
+"""
+Scanner Confidence (Numerical):
+    'Confirmed': 1,    # Certain
+    'High': 3,         # Firm
+    'Medium': 4,       # Firm
+    'Low': 6,          # Tentative
+    'Experimental': 7, # Tentative
+    'Unknown': 8,      # Tentative
+    'Ignore': 10,      # Tentative
+Numerical Severity:
+    'Critical': S0
+    'High': S1
+    'Medium': S2
+    'Low': S3
+    'Info': S4
+    not above: S5
+"""
+
+
+class TestGitlabDastParser(TestCase):
+    def test_parse_file_with_no_vuln_has_no_findings(self):
+        testfile = open("dojo/unittests/scans/gitlab_dast/gitlab_dast_zero_vul.json")
+        parser = GitlabDastParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(0, len(findings))
+
+    def test_parse_file_with_one_vuln_has_one_finding(self):
+        testfile = open("dojo/unittests/scans/gitlab_dast/gitlab_dast_one_vul.json")
+        parser = GitlabDastParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(1, len(findings))
+        finding = findings[0]
+
+        # endpoint validation
+        for endpoint in finding.unsaved_endpoints:
+            endpoint.clean()
+
+        self.assertEqual(
+            "5ec00bbc-2e53-44cb-83e9-3d35365277e3", finding.unique_id_from_tool
+        )
+        self.assertEqual(3, finding.scanner_confidence)
+        # vulnerability does not have a name: fallback to using id as a title
+        self.assertEqual("5ec00bbc-2e53-44cb-83e9-3d35365277e3", finding.title)
+        self.assertIsInstance(finding.description, str)
+
+        date = finding.date.strftime("%Y-%m-%dT%H:%M:%S.%f")
+        self.assertEqual("2021-04-23T15:46:40.615000", date)
+        self.assertIsNone(finding.references)  # should be None as there are no links
+
+        self.assertEqual("High", finding.severity)
+        self.assertEqual("", finding.mitigation)  # no solution proposed
+
+        self.assertEqual(359, finding.cwe)
+        self.assertEqual(10062, finding.cve)
+
+    def test_parse_file_with_multiple_vuln_has_multiple_findings(self):
+        testfile = open("dojo/unittests/scans/gitlab_dast/gitlab_dast_many_vul.json")
+        parser = GitlabDastParser()
+        findings = parser.get_findings(testfile, Test())
+
+        self.assertEqual(10, len(findings))
+
+        # endpoint validation
+        for finding in findings:
+            for endpoint in finding.unsaved_endpoints:
+                endpoint.clean()
+
+        # the first one is done above
+        finding = findings[1]
+        # must-have fields
+        self.assertEqual(3, finding.scanner_confidence)
+        self.assertTrue("Content Security Policy (CSP)" in finding.description)
+        self.assertEqual(10038, finding.cve)
```

Comment:
CVE is a string of this form "CVE-2021-3502". this data will break in the upper layers

Reply:
I'm not sure what did you mean by this? I thought the CVE of the above finding is the second vulnerability, i.e. findings[1], or am I missing something here?

Reply:
`cve` attribute implement this data: https://cve.mitre.org/about/faqs.html#what_is_cve_id

Reply:
I actually misunderstood the cve given by GitLab lol. It's not CVE as in the website above: it's GitLab's own vulnerability id (which is also now deprecated).

I will remove them from the parser as they aren't what I think they are.